highfidelity/hifi
# Supporting text effects + fonts on Text Entities

https://highfidelity.atlassian.net/browse/DEV-335

http://roadmap.highfidelity.com/open-source-project-proposals/p/expose-more-options-for-text-entities

Test plan:
1. Run this script: https://gist.githubusercontent.com/HifiExperiments/a855fcf3eb4c25cb3805700bea092b8f/raw/a868dadac2c68c6d3b1da6898bed0037df1eb667/TextEffectTest.js
2.  You'll see:
![image](https://user-images.githubusercontent.com/53453710/62163793-fc288880-b2cf-11e9-8be2-9fdd9ee2f0e3.png)
3. Press "o" to toggle the textAlpha.  The text should still be colored correctly (in master, it turns black, it'll look more grey now).  Press "o" again to go back to opaque.
4. Press "u" to cycle through the fonts.  In order, the fonts will go: Roboto, Courier, Timeless, Inconsolata, Roboto (invalid font), Roboto (empty font), Timeless (but downloaded from an http link)
5.  Press "i" to cycle through the text effects: none, outline, outline fill, shadow.  You'll see the behavior at various textEffectThickness:
![image](https://user-images.githubusercontent.com/53453710/62163889-32fe9e80-b2d0-11e9-92e8-669cf2c265f9.png)
![image](https://user-images.githubusercontent.com/53453710/62163900-3a25ac80-b2d0-11e9-81c8-4c2383bae23e.png)
![image](https://user-images.githubusercontent.com/53453710/62163914-40b42400-b2d0-11e9-9699-8998928d13b0.png)
The outline effect suffers from aliasing due to the thin lines, but the effect improves as you get closer or use larger textEffectThicknesses:
![image](https://user-images.githubusercontent.com/53453710/62164003-6c370e80-b2d0-11e9-9900-30cd13434838.png)
I've also left comments in the code with recommendations on how to improve that aliasing in the future.
